netaddr.go

```diff
@@ -399,10 +373,10 @@ func (ip IP) WithZone(zone string) IP {
 // If ip is the zero value, it will return false.
 func (ip IP) IsLinkLocalUnicast() bool {
 	if ip.Is4() {
-		return ip.a[o4+0] == 169 && ip.a[o4+1] == 254
+		return uint16(ip.lo>>16) == 0xa9fe
 	}
 	if ip.Is6() {
-		return ip.a[0] == 0xfe && ip.a[1] == 0x80
+		return uint16(ip.hi>>48) == 0xfe80
```

Comment:
If we do octet accessor methods (assuming they inline fine), I'd use it for both bytes here separately, even if it's a bit slower. I don't think this'll be a hot spot and I'd prefer to not have a `uint16(..>> 48)`

Reply:
Based on the lack of operation coalescing I saw when doing BigEndian stuff, I expect this will degrade back to long sequences of bytewise operations, which is a pity given that we've gone to all this trouble to make the representation avoid that.

Unfortunately I can't verify what the compiler does, because `GOSSAFUNC` on any method that uses v4() or v6() seems to crash the compiler, hilariously.

Reply:
If you can crash the compiler using GOSSAFUNC, please file an upstream issue! Those can be leading indicators of other bugs.

BTW, for the future you can get assembly out with -gcflags=-S, although that’ll be for the whole package, and it’ll get dumped to stdout.

Reply:
David Chase looks to be on it already, otherwise would have filed, yes.